packages/ra-ui-materialui/src/button/CreateButton.tsx

```diff
@@ -38,15 +38,17 @@ const CreateButton = ({
 }) => {
     const classes = useStyles({ classes: classesOverride });
     const translate = useTranslate();
-    const isSmall = useMediaQuery(theme => theme.breakpoints.down('sm'));
+    const isSmall = useMediaQuery((theme: Theme) =>
+        theme.breakpoints.down('sm')
+    );
     return isSmall ? (
         <Fab
             component={Link}
             color="primary"
             className={classnames(classes.floating, className)}
             to={`${basePath}/create`}
             aria-label={label && translate(label)}
-            {...rest}
+            {...rest as any}
```

Comment:
I had to disable type checking here because of a material-ui bug: `Button` doesn't recognize the `component` prop :/

See https://github.com/mui-org/material-ui/issues/15827